.buildkite/pipeline.yml

```diff
@@ -4,375 +4,435 @@ agents:
+        ./scripts/buildkite/main/bench-latency.sh
+      depends_on: trigger-benchmarks
+      timeout_in_minutes: 20
+      agents:
+        system: x86_64-linux
+        queue: adrestia-bench
+      artifact_paths: [ "./bench-results.csv" ]
+      concurrency: 4
+      concurrency_group: 'concurrent-benchmarks'
+
+    - label: DB Benchmark (linux)
+      command: |
+        export BENCHMARK_CSV_FILE="`pwd`/bench-results.csv"
+        ./scripts/buildkite/main/bench-db.sh
+      depends_on: trigger-benchmarks
+      timeout_in_minutes: 50
+      agents:
+        system: x86_64-linux
+        queue: adrestia-bench
+      artifact_paths: [ "./bench-results.csv" ]
+      concurrency: 4
+      concurrency_group: 'concurrent-benchmarks'
+
+    - label: Read-blocks Benchmark (linux)
+      command: |
+        export BENCHMARK_CSV_FILE="`pwd`/bench-results.csv"
+        ./scripts/buildkite/main/bench-read-blocks.sh
+      depends_on: trigger-benchmarks
+      timeout_in_minutes: 20
+      agents:
+        system: x86_64-linux
+        queue: adrestia-bench
+      artifact_paths: [ "./bench-results.csv" ]
+      concurrency: 4
+      concurrency_group: 'concurrent-benchmarks'
+
+    - label: Memory Benchmark (linux)
+      command: |
+        export BENCHMARK_CSV_FILE="`pwd`/bench-results.csv"
+        ./scripts/buildkite/main/bench-memory.sh
+      depends_on: trigger-benchmarks
+      timeout_in_minutes: 20
+      agents:
+        system: x86_64-linux
+        queue: adrestia-bench
+      artifact_paths: [ "./bench-results.csv" ]
+      concurrency: 4
+      concurrency_group: 'concurrent-benchmarks'
+
+    - block: Restoration Benchmark (linux)
+      if: build.env("RELEASE_CANDIDATE") == null
+      depends_on: linux-nix
+      key: restore-block
+
+    - label: Restore benchmark (linux)
+      command: "./scripts/buildkite/main/bench-restore.sh mainnet"
+      depends_on: restore-block
+      timeout_in_minutes: 1200
+      agents:
+        system: ${linux}
+        queue: adrestia-bench
+      concurrency: 1
+      concurrency_group: 'restoration-benchmark'
+
+  - group: MacOS Artifacts
+    key: macos-artifacts
+    steps:
+
+    - label: Check Nix (macOS)
+      key: macos-nix
+      commands:
+        - 'nix flake info'
+      agents:
+        system: ${macos}
+
+    - block: MacOS Steps
+      if: |
+        build.branch !~ /^gh-readonly-queue\/master/
+          && build.branch != "master"
+          && build.env("RELEASE_CANDIDATE") == null
+      depends_on: macos-nix
+      key: block-macos
+
+    - label: Run Unit Tests (macOS, x86_64)
+      key: macos-intel-tests-run-unit
+      depends_on: block-macos
+      command: 'nix build -L .#ci.x86_64-darwin.tests.run.unit'
+      agents:
+        system: ${macos}
+
+    - label: Run Unit Tests (macOS, arm64)
+      key: macos-arm64-tests-run-unit
+      depends_on: block-macos
+      command: 'nix build -L .#ci.aarch64-darwin.tests.run.unit'
+      agents:
+        system: ${macos}
+
```

Comment:
To be consistent with the nomenclature, I recommend moving the unit tests to the "MacOS Checks" section.